### PR TITLE
Update Helm chart to reflect different Ingress api versions

### PR DESCRIPTION
**Pull request by dirien**
Signed-off-by: Engin Diri <engin.diri@mail.schwarz>

## Motivation and Context

Fixex #674
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

started k8s cluster (1.17) with rancher-desktiop
```bash
kubectl version         
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.17", GitCommit:"68b4e26caf6ede7af577db4af62fb405b4dd47e6", GitTreeState:"clean", BuildDate:"2021-03-18T01:02:41Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.17+k3s1", GitCommit:"e8c9484078bc59f2cd04f4018b095407758073f5", GitTreeState:"clean", BuildDate:"2021-01-14T22:16:57Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

 helm upgrade -i  checkly-operator . --set ingress.enabled=true --create-namespace --namespace openfaas-fn

kubectl get  ingress openfaas-ingress  -n openfaas-fn  -o yaml
```

```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    meta.helm.sh/release-name: checkly-operator
    meta.helm.sh/release-namespace: openfaas-fn
  creationTimestamp: "2021-07-16T15:17:30Z"
  generation: 1
  labels:
    app: openfaas
    app.kubernetes.io/managed-by: Helm
    chart: openfaas-7.3.2
    heritage: Helm
  name: openfaas-ingress
  namespace: openfaas-fn
  resourceVersion: "333"
  selfLink: /apis/extensions/v1beta1/namespaces/openfaas-fn/ingresses/openfaas-ingress
  uid: 9a9e45ab-d479-4b6e-bbb6-e78c33d988cc
spec:
  rules:
  - host: gateway.openfaas.local
    http:
      paths:
      - backend:
          serviceName: gateway
          servicePort: 8080
        path: /
status:
  loadBalancer: {}

```

Change to 1.21.1 of kubernetes

```bash
kubectl version         
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:52:14Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2+k3s1", GitCommit:"5a67e8dc473f8945e8e181f6f0b0dbbc387f6fca", GitTreeState:"clean", BuildDate:"2021-06-21T20:52:44Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}

helm upgrade -i  checkly-operator . --set ingress.enabled=true --create-namespace --namespace openfaas-fn
```
```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    meta.helm.sh/release-namespace: openfaas-fn
  creationTimestamp: "2021-07-16T14:51:56Z"
  generation: 1
  labels:
    app: openfaas
    app.kubernetes.io/managed-by: Helm
    chart: openfaas-7.3.2
    heritage: Helm
    release: checkly-operator
  name: openfaas-ingress
  namespace: openfaas-fn
  resourceVersion: "1154"
  uid: 88a10769-9670-43d8-8558-2edc8442f35b
spec:
  rules:
  - host: gateway.openfaas.local
    http:
      paths:
      - backend:
          service:
            name: gateway
            port:
              number: 8080
        path: /
        pathType: ImplementationSpecific
status:
  loadBalancer: {}

```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [ x I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: chart/openfaas/templates/ingress.yaml [hunk 1]
```diff
@@ -21,8 +22,15 @@ spec:
         paths:
           - path: {{ $host.path }}
             backend:
+              {{- if $apiIsStable }}
```
Comment (alexellis) [inline chart/openfaas/templates/ingress.yaml:25]: Oh wow, there is even more to change than I had remembered.. 

I would love a quick blog post on this recipe on openfaas.com, if you're open to writing it as a guest author?

**Review by alexellis — DISMISSED** [review]
LGTM

File: chart/openfaas/values.yaml [hunk 2]
```diff
@@ -180,6 +182,10 @@ nats:
 # ingress configuration
 ingress:
   enabled: false
+  ## For k8s >= 1.18 you need to specify the pathType
+  ## See https://kubernetes.io/blog/2020/04/02/improvements-to-the-ingress-api-in-kubernetes-1.18/#better-path-matching-with-path-types
+  pathType: ImplementationSpecific
```
Comment (dirien) [inline chart/openfaas/values.yaml:187]: @alexellis : Should we enable pathType or comment it out?

File: chart/openfaas/templates/ingress.yaml (hunk 1 above)
Comment (dirien) [reply chart/openfaas/templates/ingress.yaml:30]: yeah, on top with 1.18 comes pathType :D 

I know the pain, we have OCP 3.11 at work and now 4.7 :D Migrating is my second hobby

File: chart/openfaas/values.yaml (hunk 2 above)
Comment (alexellis) [reply chart/openfaas/values.yaml:187]: I am not sure that it's required for this part of the chart, so let's go with less or more. It's easily reversible.

File: chart/openfaas/values.yaml [hunk 3]
```diff
@@ -201,16 +207,16 @@ ingressOperator:
...
       memory: "25Mi"
 
-nodeSelector: {}
+nodeSelector: { }
 
-tolerations: []
+tolerations: [ ]
 
-affinity: {}
+affinity: { }
```
Comment (alexellis) [inline chart/openfaas/values.yaml:214]: Can you change `{ }` back to `{}` which is what I'm used to seeing in every other helm chart.

Was this a linter that you used?

**Review by alexellis — APPROVED** [review]
Approved, but left comments.

File: chart/openfaas/values.yaml (hunk 2 above)
Comment (dirien) [reply chart/openfaas/values.yaml]: ok commented it out

File: chart/openfaas/values.yaml (hunk 3 above)
Comment (dirien) [reply chart/openfaas/values.yaml]: yep done...  i use  intellj

**Review by alexellis — APPROVED** [review]
LGTM

**Pull request merged**